libcxx/include/module.modulemap

```diff
@@ -1229,7 +1229,11 @@ module std [system] {
   module filesystem {
     module copy_options                   { header "__filesystem/copy_options.h" }
     module directory_entry                { header "__filesystem/directory_entry.h" }
-    module directory_iterator             { header "__filesystem/directory_iterator.h" }
+    module directory_iterator             {
+      header "__filesystem/directory_iterator.h"
+      // This is a workaround for https://github.com/llvm/llvm-project/issues/120108.
+      export *
```

Comment:
Did you try to use more targeted exports?
Looking at the header maybe `export std.filesystem.directory_entry` may work as well since that header exports `__dir_element_proxy`.

Reply:
For this, we want to change visibility of `std::filesystem::__dir_stream`, but we cannot export specific class in code by `export` annotation?

Reply:
I tried
```
    module directory_entry                {
      header "__filesystem/directory_entry.h"
      export *
    }
    module directory_iterator             {
      header "__filesystem/directory_iterator.h"
      export std.filesystem.directory_entry
    }
```
but that didn't fix symbol's visibility.

Reply:
Thanks for testing.